prog/weaveproxy/main.go

```diff
@@ -15,6 +15,23 @@ var (
 	defaultListenAddrs = []string{"tcp://0.0.0.0:12375", "unix:///var/run/weave.sock"}
 )
 
+type listOpts struct {
+	values *[]string
```

Comment:
Good point! I was adapting from the package Paul mentions above, and didn't flatten the single-field struct.